crates/cli/cli/src/lib.rs

```diff
@@ -42,4 +44,22 @@ pub trait RethCli: Sized {
     {
         <Self as Parser>::try_parse_from(itr)
     }
+
+    /// Executes a command.
+    fn with_runner<F: FnOnce(Self, CliRunner)>(self, f: F) {
+        let runner = CliRunner::default();
+
+        f(self, runner)
+    }
+
+    /// Parses and executes a command.
+    fn execute<F: FnOnce(Self, CliRunner)>(f: F) -> Result<(), Error>
+    where
+        Self: Parser + Sized,
+    {
+        let cli = Self::parse_args()?;
+        cli.with_runner(f);
+
+        Ok(())
+    }
```

Comment:
these need another generic for the return type of `F`, so these functions should return -> R